src/platform/graphics/graphics-device.js

```diff
@@ -444,7 +454,8 @@ class GraphicsDevice extends EventHandler {
     }
 
     updateClientRect() {
-        this.clientRect = this.canvas.getBoundingClientRect();
+        this.clientRect.width = this.canvas.width;
+        this.clientRect.height = this.canvas.height;
```

Comment:
My guess is that the `canvas.width` and `canvas.height` properties are the pixel resolution of the canvas, whereas `getBoundingClientRect` would return the CSS width and height. So I would imagine `CameraComponent#worldToScreen` and `CameraComponent#screenToWorld` would break. Maybe needs `this.canvas.style.width` instead? So `parseInt(this.canvas.style.width, 10)`. But I'm not sure if that would work in all circumstances...

Reply:
Unfortunately it would not work. Because Web Workers don't do the page layout, they are unaware of `el.getBoundingClientRect()`, `el.style`, `getComputedStyle(el)`. Only the main/rendering thread knows about layout dimensions and coordinates of mouse button clicks.

Therefore I can think right now of the following independent options:

1. Change `CameraComponent#screenToWorld` and `CameraComponent#worldToScreen` to accept normalized coordinates for the mouse clicks. i.e. let the user do the division of the mouse click coordinates by the canvas screen dimensions, of which they have full knowledge.
2. Pass the canvas screen dimensions into the `Application` constructor.
3. Leave the implementation as it is and require that `OffscreenCanvas` not be scaled by CSS when `CameraComponent#screenToWorld` and `CameraComponent#worldToScreen` are to be used.

I can't judge what would be suitable for the project, but option 3 would be least amount of work while still being correct for the general use case.

Reply:
option 3 seems to be the best fit for now. also, may I enquiry if there's anything else prevent this from being merged to main?

Reply:
@willeastcott Is there still interest in merging this PR? I do need this feature and I think it would be generally useful.

I rebased the PR it on _main_ and addressed option 3 by doing a conditional calculation of the _clientRect_ to not break other existing features.

Reply:
definitely would like to this get merged